File: redditClone/userservice/src/main/java/io/service/user/userservice/services/UserService.java
package io.service.user.userservice.services;

import io.service.user.userservice.core.utilities.results.DataResult;
import io.service.user.userservice.core.utilities.results.Result;
import io.service.user.userservice.entities.User;
import io.service.user.userservice.entities.dtos.userSaveRequestDto;

import java.util.List;

public interface UserService {
    Result add(userSaveRequestDto user);
    DataResult<List<String>> findByUsername(String username);

    DataResult<List<User>> findByUsernames(List<Integer> users);

    DataResult MainLatestPosts(int userid);

    DataResult getById(List<Integer> userId);
}

File: reddit-clone-frontend/src/services/userService.js
import axios from "axios";
import { api } from "../App";

export default class UserService{
    getUserById(userId){
        return axios.get(api+"user/getUser?userId="+userId)
    }

    getMainLatestPosts(userId){
        return axios.get(api+"user/MainLatestPosts?userid="+userId)
    }
}